craft_parts/parts.py

```diff
@@ -63,6 +63,18 @@ class Config:
         allow_mutation = False
         alias_generator = lambda s: s.replace("_", "-")  # noqa: E731
 
+    # pylint: disable=no-self-argument,no-self-use
+    @validator("stage_files", "prime_files", each_item=True)
+    def validate_relative_path_list(cls, item):
+        """Check if the list does not contain empty of absolute paths."""
+        assert item != "", f"{item!r} must be a relative path (cannot be empty)"
```

Comment:
The text between parentesis should be to explain/exemplify the other part. In this case they are two different reasons (both valid). What about "{} must be a relative path and cannot be empty", or "{} must be a relative non-empty path"? Or maybe, as below it's tested the "relativeness" itself, what about "{} path cannot be empty"?